### PR TITLE
Bibliographic lists extraction

### PR DESCRIPTION
**Pull request by Memmo91**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -77,6 +78,7 @@ export function initializeApp(appConfig: AppConfig) {
     TextVersionsComponent,
     VersionPanelComponent,
     WitnessPanelComponent,
+    BibliographyComponent,
```
Comment (szenzaro) [inline src/app/app.module.ts]: Reorder alphabetically

File: src/app/app.module.ts [hunk 2]
```diff
@@ -43,6 +43,7 @@ import { PageComponent } from './components/page/page.component';
 import { PinboardComponent } from './pinboard/pinboard.component';
 import { PinnerComponent } from './pinboard/pinner/pinner.component';
 import { PinboardPanelComponent } from './panels/pinboard-panel/pinboard-panel.component';
+import { BibliographyComponent } from './components/bibliography/bibliography.component';
```
Comment (szenzaro) [inline src/app/app.module.ts]: Keep alphabetical order

File: src/app/app.component.html [hunk 3]
```diff
@@ -1,3 +1,4 @@
+<evt-bibliography></evt-bibliography>
```
Comment (szenzaro) [inline src/app/app.component.html]: Should this component be visualised here?

File: src/app/components/bibliography/bibliography.component.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+            <select>
+            
+            </select>
+            <select>
+            
+            </select>
+        </span>
+    </div>
+</div>
+
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.html]: This code does nothing. Why it is here?

File: src/app/components/bibliography/bibliography.component.scss [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    cursor: pointer;
+    display: none; 
+	overflow:scroll;
+    overflow-x:hidden;
+    background-color: #fefefe;
+    margin: auto;
+    padding: 20px;
+    border: 1px solid #888;
+    width: 600px;
+    // height: 500px;
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.scss]: Remove commented code.

File: src/app/components/bibliography/bibliography.component.scss [hunk 6]
```diff
@@ -0,0 +1,14 @@
+
+.biblSpace{
+    cursor: pointer;
+    display: none; 
+	overflow:scroll;
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.scss]: Indent correctly

File: src/app/components/bibliography/bibliography.component.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+  selector: 'evt-bibliography',
+  templateUrl: './bibliography.component.html',
+  styleUrls: ['./bibliography.component.scss']
+})
+export class BibliographyComponent implements AfterViewInit {
+
+  constructor( ) {
+    const xmlFile = 'assets/data/edition.xml';
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: Edition path should be the one defined in the config file. See `AppConfig`

File: src/app/components/bibliography/bibliography.component.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    const xhttp = new XMLHttpRequest();
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: No need to use `XMLHttpRequest`. See `AppConfig`

File: src/app/components/bibliography/bibliography.component.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    xhttp.open('GET', xmlFile, true);
+    xhttp.send();
+    xhttp.onreadystatechange = function() {
+      if (this.readyState === 4 && this.status === 200) {
+        const parser = new DOMParser();
+        const xmlDoc = parser.parseFromString(this.response, 'text/xml');
+        const listsBibl = Array.from(xmlDoc.getElementsByTagName('listBibl'));
+        console.log(listsBibl.length);
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: no console logs

File: src/app/components/bibliography/bibliography.component.ts [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        listsBibl.forEach((l) => {
+          l.childNodes.forEach((field) => {
+            if (!field.nodeValue) {
+              console.log(field);
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: no console logs

File: src/app/components/bibliography/bibliography.component.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+            }
+          });
+        });
+      }
+    };
+  }
+
+  ngAfterViewInit() {
+    (document.querySelectorAll('.cloosingRood')[0] as HTMLElement).onclick = () => {
+      (document.querySelectorAll('.biblSpace')[0] as HTMLElement).style.display = 'none';
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: Why are you using this hack to hide the component you are trying to display?

File: src/app/main-menu/main-menu.component.ts [hunk 12]
```diff
@@ -91,6 +91,7 @@ export class MainMenuComponent implements OnInit, OnDestroy {
 
   private openGlobalDialogInfo() {
     // TODO openGlobalDialogInfo
+    (document.querySelectorAll('.biblSpace')[0] as HTMLElement).style.display = 'block';
```
Comment (szenzaro) [inline src/app/main-menu/main-menu.component.ts]: ???

File: src/app/components/bibliography/bibliography.component.ts [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    xhttp.open('GET', xmlFile, true);
+    xhttp.send();
+    xhttp.onreadystatechange = function() {
+      if (this.readyState === 4 && this.status === 200) {
+        const parser = new DOMParser();
+        const xmlDoc = parser.parseFromString(this.response, 'text/xml');
+        const listsBibl = Array.from(xmlDoc.getElementsByTagName('listBibl'));
+        console.log(listsBibl.length);
+        listsBibl.forEach((l) => {
+          l.childNodes.forEach((field) => {
```
Comment (szenzaro) [inline src/app/components/bibliography/bibliography.component.ts]: Where do you extract the information related to listBibl?

File: src/app/app.component.html (hunk 3 above)
Comment (Memmo91) [reply src/app/app.component.html]: For now, yes.

File: src/app/components/bibliography/bibliography.component.ts (hunk 11 above)
Comment (Memmo91) [reply src/app/components/bibliography/bibliography.component.ts]: it seems simple to read, and not too bulky

File: src/app/main-menu/main-menu.component.ts (hunk 12 above)
Comment (Memmo91) [reply src/app/main-menu/main-menu.component.ts]: Same as up.

File: src/app/components/bibliography/bibliography.component.ts (hunk 11 above)
Comment (szenzaro) [reply src/app/components/bibliography/bibliography.component.ts]: You should use the angular framework instead of polluting the code so badly

**Pull request closed without merging**